Loop Status Extension/StatusExtensionContext.swift

```diff
@@ -0,0 +1,156 @@
+//
+//  StatusExtensionContext.swift
+//  Loop Status Extension
+//
+//  Created by Bharat Mediratta on 11/25/16.
+//  Copyright © 2016 LoopKit Authors. All rights reserved.
+//
+//  This class allows Loop to pass context data to the Loop Status Extension.
+
+import Foundation
+import HealthKit
+import LoopKit
+import InsulinKit
+import MinimedKit
+
+struct ReservoirContext: ReservoirValue {
+    var startDate: Date
+    var unitVolume: Double
+    var capacity: Int
+}
+
+struct LoopContext {
+    var dosingEnabled: Bool
+    var lastCompleted: Date?
+}
+
+struct NetBasalContext {
+    var rate: Double
+    var percentage: Double
+    var startDate: Date
+}
+
+struct SensorDisplayableContext: SensorDisplayable {
+    var isStateValid: Bool
+    var stateDescription: String
+    var trendType: GlucoseTrend?
+    var isLocal: Bool
+}
+
+struct GlucoseContext {
+    var latest: GlucoseValue
+    var sensor: SensorDisplayable?
+}
+
+final class StatusExtensionContext: NSObject, RawRepresentable {
+    typealias RawValue = [String: Any]
+    private let version = 1
+    
+    var latestGlucose: GlucoseContext?
+    var reservoir: ReservoirContext?
+    var loop: LoopContext?
+    var netBasal: NetBasalContext?
+    var batteryPercentage: Double?
+    var eventualGlucose: String?
+    
+    override init() {
+        super.init()
+    }
+    
+    required init?(rawValue: RawValue) {
+        super.init()
+        let raw = rawValue
+        
+        if let unitString = raw["latestGlucose_unit"] as? String,
+           let latestValue = raw["latestGlucose_value"] as? Double,
+           let startDate = raw["latestGlucose_startDate"] as? Date {
+            latestGlucose = GlucoseContext(
```

Comment:
yeah, I toyed with that because I would very much like to use nested dictionaries. But then I wound up with less namespacing issues in the strings, but more boilerplate code to implement RawRepresentable in each class. I figured I'd go for less code, but happy to do it with nested RawRepresentable dicts if you think it's the right way to go.